### PR TITLE
Use install_modules_dependencies

### PR DESCRIPTION
**Pull request by piaskowyk**
## Summary

Based on the changes in our libraries mentioned in this PR (https://github.com/software-mansion/react-native-gesture-handler/pull/2635), we need to make adjustments to our `.podspec` file. We should utilize the `install_modules_dependencies` feature available in React Native since version 0.72.

Backward compatibility:
image

## Test plan

Check CI

### CODE REVIEW

File: FabricExample/ios/Podfile.lock [hunk 1]
```diff
@@ -1514,4 +1500,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 200355ebe627c91121916eeac614c78c24ec30b7
 
-COCOAPODS: 1.14.0
+COCOAPODS: 1.12.1
```
Comment (tomekzaw) [inline FabricExample/ios/Podfile.lock:1503]: `brew upgrade cocoapods` 😄

File: Example/ios/ReanimatedExample.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -650,13 +650,7 @@
 					"-DFOLLY_MOBILE=1",
 					"-DFOLLY_USE_LIBCPP=1",
 				);
-				OTHER_LDFLAGS = (
-					"$(inherited)",
-					"-Wl",
-					"-ld_classic",
-					" ",
-					"-Wl -ld_classic ",
-				);
```
Comment (tomekzaw) [inline Example/ios/ReanimatedExample.xcodeproj/project.pbxproj:659]: upgrade to Xcode 15

File: RNReanimated.podspec [hunk 3]
```diff
@@ -6,67 +6,32 @@ config = find_config()
...
 boost_compiler_flags = '-Wno-documentation'
-fabric_flags = fabric_enabled ? '-DRCT_NEW_ARCH_ENABLED' : ''
+fabric_flags = new_arch_enabled ? '-DRCT_NEW_ARCH_ENABLED' : ''
 example_flag = config[:is_reanimated_example_app] ? '-DIS_REANIMATED_EXAMPLE_APP' : ''
 version_flag = '-DREANIMATED_VERSION=' + reanimated_package_json["version"]
 debug_flag = is_release ? '-DNDEBUG' : ''
 ios_min_version = config[:react_native_minor_version] >= 73 ? '10.0' : '9.0'
+if new_arch_enabled
+  ios_min_version = '11.0'
+end
```
Comment (tomekzaw) [inline RNReanimated.podspec:23]:
```suggestion
ios_min_version = new_arch_enabled ? '11.0' : config[:react_native_minor_version] >= 73 ? '10.0' : '9.0'
```

File: RNReanimated.podspec [hunk 4]
```diff
@@ -97,8 +62,53 @@ Pod::Spec.new do |s|
...
+  
+  s.name         = "RNReanimated"
+  s.version      = reanimated_package_json["version"]
+  s.summary      = reanimated_package_json["description"]
+  s.description  = <<-DESC
+                  RNReanimated
+                   DESC
+  s.homepage     = "https://github.com/software-mansion/react-native-reanimated"
+  s.license      = "MIT"
+  # s.license    = { :type => "MIT", :file => "FILE_LICENSE" }
```
Comment (tomekzaw) [inline RNReanimated.podspec:81]:
```suggestion
```

File: RNReanimated.podspec [hunk 5]
```diff
@@ -97,8 +62,53 @@ Pod::Spec.new do |s|
...
+    "FRAMEWORK_SEARCH_PATHS" => "\"${PODS_CONFIGURATION_BUILD_DIR}/React-hermes\"",
+    "CLANG_CXX_LANGUAGE_STANDARD" => "c++17",
+  }
+  s.compiler_flags = folly_compiler_flags + ' ' + boost_compiler_flags + ' -DHERMES_ENABLE_DEBUGGER'
+  s.xcconfig = {
+    "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/boost\" \"$(PODS_ROOT)/boost-for-react-native\" \"$(PODS_ROOT)/glog\" \"$(PODS_ROOT)/RCT-Folly\" \"$(PODS_ROOT)/Headers/Public/React-hermes\" \"$(PODS_ROOT)/Headers/Public/hermes-engine\" \"$(PODS_ROOT)/#{config[:react_native_common_dir]}\"",
+    "OTHER_CFLAGS" => "$(inherited)" + " " + folly_flags + " " + fabric_flags + " " + example_flag + " " + version_flag + " " + debug_flag
+  }
+  s.requires_arc = true
+  s.dependency "ReactCommon/turbomodule/core"
```
Comment (tomekzaw) [inline RNReanimated.podspec:108]: do we need this dependency in both cases?

Comment (piaskowyk) [reply RNReanimated.podspec:108]: Yes, we need it :/

Comment (tomekzaw) [reply RNReanimated.podspec:104]: 😢

File: RNReanimated.podspec [hunk 6]
```diff
@@ -6,67 +6,29 @@ config = find_config()
...
-    "CLANG_CXX_LANGUAGE_STANDARD" => "c++17",
-  }
-  s.compiler_flags = folly_compiler_flags + ' ' + boost_compiler_flags + ' -DHERMES_ENABLE_DEBUGGER'
-  s.xcconfig = {
-    "HEADER_SEARCH_PATHS" => "\"$(PODS_ROOT)/boost\" \"$(PODS_ROOT)/boost-for-react-native\" \"$(PODS_ROOT)/glog\" \"$(PODS_ROOT)/RCT-Folly\" \"$(PODS_ROOT)/Headers/Public/React-hermes\" \"$(PODS_ROOT)/Headers/Public/hermes-engine\" \"$(PODS_ROOT)/#{config[:react_native_common_dir]}\"",
-    "OTHER_CFLAGS" => "$(inherited)" + " " + folly_flags + " " + fabric_flags + " " + example_flag + " " + version_flag + " " + debug_flag
-  }
-
-  s.requires_arc = true
+ios_min_version = new_arch_enabled ? '11.0' : config[:react_native_minor_version] >= 73 ? '10.0' : '9.0'
```
Comment (tomekzaw) [inline RNReanimated.podspec:20]: do we want to add backward compatibility for 0.72?

File: RNReanimated.podspec [hunk 7]
```diff
@@ -6,67 +6,29 @@ config = find_config()
...
 
+module LegacyModule
+def self.install_modules_dependencies(s, config, new_arch_enabled, using_hermes)
```
Comment (tomekzaw) [inline RNReanimated.podspec:23]: do we want to introduce a level of indentation here?

File: RNReanimated.podspec [hunk 8]
```diff
@@ -2,22 +2,75 @@ require "json"
...
+      s.dependency 'React-RCTAppDelegate'
+    end
+  end
+
+  def self.get_ios_min_version()
+    if $config[:react_native_minor_version] <= 72
+      return '9.0'
+    end
+    # 0.73+ supports iOS 13.4+, https://github.com/facebook/react-native/pull/39478
+    return '13.4'
```
Comment (tomekzaw) [inline RNReanimated.podspec:71]: Let's always use `>=` when comparing RN versions starting from the newest to oldest
```suggestion
    if $config[:react_native_minor_version] >= 73
      return '13.4'
    else
      return '9.0'
    end
```

File: RNReanimated.podspec [hunk 9]
```diff
@@ -2,22 +2,68 @@ require "json"
...
-example_flag = config[:is_reanimated_example_app] ? '-DIS_REANIMATED_EXAMPLE_APP' : ''
+fabric_flags = $new_arch_enabled ? '-DRCT_NEW_ARCH_ENABLED' : ''
+example_flag = $config[:is_reanimated_example_app] ? '-DIS_REANIMATED_EXAMPLE_APP' : ''
 version_flag = '-DREANIMATED_VERSION=' + reanimated_package_json["version"]
 debug_flag = is_release ? '-DNDEBUG' : ''
-ios_min_version = config[:react_native_minor_version] >= 73 ? '10.0' : '9.0'
+ios_min_version = $config[:react_native_minor_version] >= 73 ? '13.4' : '9.0'
+
+module LegacyModule
+  def self.install_modules_dependencies(s)
```
[the remark this answers is not in the document]
Comment (piaskowyk) [reply RNReanimated.podspec]: ok, I found the way - now we uses regular function 🎉

**Comment by genesiscz** [comment]
This broke our app - Your podspec is overriding our `$config` variable which caused post-install to fail. I am not sure if it's something you care about, but letting you know so you are aware of that :-)

**Comment by tomekzaw** [comment]
@genesiscz Thanks for letting us know, we'll discuss how we can improve it. I hope you're the only one who's affected :)

**Pull request merged**